src/protocol/get_pfm_id.rs

```diff
@@ -0,0 +1,214 @@
+        let pfm_valid = PFMValid::from_wire(r, arena)?;
+        let identifier = IdentifierType::from_wire(r, arena)?;
+        let pfm_id = match identifier {
+            IdentifierType::Version => {
+                let mut version: [u8; 4] = [0; 4];
+                r.read_bytes(&mut version)?;
+                Identifier::Version {id: version}
+            }
+            IdentifierType::Platform => {
+                if r.remaining_data() == 0 {
+                    Identifier::Platform {id: &[]}
+                } else {
+                    let bytes = r.read_direct(
+                    arena,
+                    Layout::from_size_align(
+                        r.remaining_data(),
+                        core::mem::align_of::<u8>())
+                    .expect(
+                        "Layout for u8 slice can't be created. This indicates a bug in the code."))?;
+
+                    Identifier::Platform{id: bytes}
+                }
+            }
+        };
+        Ok(Self { pfm_valid, pfm_id })
+    }
+
+    fn Response::to_wire(&self, w) {
+        self.pfm_valid.to_wire(&mut w)?;
+        match self.pfm_id {
+            Identifier::Version{id: ref bytes } => {
+                let idtype = IdentifierType::Version;
+                idtype.to_wire(&mut w)?;
+                w.write_bytes(bytes)?;
+            }
+            Identifier::Platform{ id: ref bytes } => {
+                let idtype = IdentifierType::Platform;
+                idtype.to_wire(&mut w)?;
+                w.write_bytes(bytes)?;
+            }
+        }
+        Ok(())
+    }
+
+}
+
+wire_enum! {
+    /// PFM Region.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum PFMRegion: u8 {
+        /// PFM Region Active
+        Active = 0x00,
+        /// PFM Region Pending
+        Pending = 0x01,
+    }
+}
+
+wire_enum! {
+    /// The id type requested.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum IdentifierType: u8 {
+        /// PFM Version ID
+        Version = 0x00,
+        /// PFM Platform ID
+        Platform = 0x01,
+    }
+}
+
+wire_enum! {
+    /// The id type requested.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum PFMValid: u8 {
+        /// PFM is valid
+        Valid = 0x00,
+        /// PFM is invalid
+        Invalid = 0x01,
+    }
+}
+make_fuzz_safe!(PFMValid);
+
+make_fuzz_safe! {
+    /// The PFM Identifier (either platform or version).
+    #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum Identifier<'a> {
+        /// Identifier of type Version. Set when `IdentifierType::Version` is set
+        /// in request. Uses a 4-byte identifier.
+        Version {
```

Comment:
Oh, I don't support tuple variants here, do I? I think these should be tuple variants. Can you file a bug against me to add tuple variant support to this macro?

Reply:
Done #151

Reply:
Add TODO as well to clean up.